**Introduction**
- A central focus of single-cell RNA-seq is the ability to survey the diversity of cell types in a sample.
- Proteomic technologies have the advantage of assaying the final functional product of gene expression, but at the single-cell level, proteomics assays are constrained to a
limited, pre-selected repertoire of molecules, which precludes an unbiased, comprehensive analysis of cell phenotypes. 
- Assay of the transcriptome (the abundance of all transcribed RNAs) in a single cell with methodologies such as RNA-seq provides an alternative means to classify and characterize cells at the
molecular level.
- scRNA seq protocol outline
  - cDNA --> RNA --> amplifyusing PCR --> sequencing ---> Gene product quantification
- tens of cells --> 100,000 cells per study
- SMART-seq , 
- CEL-seq uses IVT
- MARS-seq
- InDrop
- Seq-well
- FACS
- STRT-seq-2i
- Drop-seq protocol - isolate cells in droplets and carry out barcoded cDNA preparation within each droplet

**Limitations**
- requirement of large volumes of cell suspensions - only fraction of cells in the suspension will be captured
- making cell suspensions is difficult
- storage of cell suspensions is challenging
- still costly
- with the suspension of cells is created, spatial context of cells is lost

**Conclusion**
- can now study cell type compositions
- 10-100,000 cells
- reporducibly cataloging cell types and gene expression patterns
